galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -181,8 +315,143 @@ def _dens(self,R,z,phi=0.,t=0.):
+        # if z is None:
+        #     r= R
+        # else:
+        #     r= sqrt(R**2.+z**2.)
+        return (r/a)**(3.-alpha)/(3.-alpha)*hyp2f1(3.-alpha,-alpha+beta,4.-alpha,-r/a)
+
+
+class DehnenSphericalPotential(TwoPowerSphericalPotential):
+    """Class that implements the Dehnen Spherical Potential from Dehnen 1993
+
+    .. math::
+
+          \\rho(r) = \\frac{\\mathrm{amp}(3-\\alpha)}{4\\,\\pi\\,a^3}\\,\\frac{1}{(r/a)^{\\alpha}\\,(1+r/a)^{4-\\alpha}}
+    """
+    def __init__(self,amp=1.,a=1.,alpha=1.5,dehnen_amp=False,
+                 normalize=False,ro=None,vo=None):
+        """
+        NAME:
+
+           __init__
+
+        PURPOSE:
+
+           initialize a Dehnen Spherical Potential
+
+        INPUT:
+
+           amp - amplitude to be applied to the potential (default: 1); can be a Quantity with units of mass or Gxmass will be rescaled to amp*(3-alpha) (except for Jaffe, which is already scaled)
+
+           a - scale radius (can be Quantity)
+
+           alpha - inner power, restricted to [0, 3)
+
+           dehnen_amp - if True, use dehnen normalization, amp -> amp * (3 - alpha)
+
+           normalize - if True, normalize such that vc(1.,0.)=1., or, if given as a number, such that the force is this fraction of the force necessary to make vc(1.,0.)=1.
+
+           ro=, vo= distance and velocity scales for translation into internal units (default from configuration file)
+
+        OUTPUT:
+
+           (none)
+
+        HISTORY:
+
+           2019-10-07 - Started - Starkman (UofT)
+
+        """
+        # instantiate
+        if dehnen_amp:
+            amp= amp*(3-alpha)  # difference between bovy and Dehnen implementation
+        super(DehnenSphericalPotential, self).__init__(
+            amp=amp,a=a,alpha=alpha,beta=4,
+            normalize=normalize,ro=ro,vo=vo
+        )
+        if ((self.__class__ == DehnenSphericalPotential) &
+            (alpha == round(alpha))):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=a,alpha=round(alpha),beta=4,
+                normalize=False,ro=None,vo=None)
+        else:
+            self.integerSelf= None
+        # set properties
+        self.hasC= True
+        self.hasC_dxdv= True
+        self._nemo_accname= 'Dehnen'
+        return None
+
+    @property
+    def alpha(self):
+        return self._alpha
+    @alpha.setter
+    def alpha(self, alpha):
+        # checking if alpha allowed to change
+        self._alpha = alpha  # setting alpha
+        # checking if should update .integerSelf
+        if alpha == round(alpha):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=self.a,alpha=round(alpha),beta=4,
+                normalize=False,)
+        else:
+            self.integerSelf= None
+
+    @property
+    def beta(self):
+        return self._beta
+    @beta.setter
+    def beta(self, value):
+        raise Exception('cannot modify beta')
```

Comment:
Isn't this handled by the super-class? Anyway, see comment about that above.

Reply:
I've commented this out.
This version doesn't permit beta to vary.